helm/templates/server/deployment.yaml

```
@@ -0,0 +1,177 @@
+        args:
+        {{- range . }}
+        - {{ . | quote }}
+        {{- end }}
+        {{- end }}
+        ports:
+          - containerPort: {{ .Values.server.service.port }}
+            protocol: TCP
+        envFrom:
+        - configMapRef:
+            name: {{ include "cpnConsole.fullname" . }}-server
+        - secretRef:
+            name: {{ include "cpnConsole.fullname" . }}-server
+        {{- if .Values.server.startupProbe.enabled }}
+        startupProbe:
+          httpGet:
+            path: {{ .Values.server.healthcheckPath }}
+            port: {{ .Values.server.container.port }}
+          initialDelaySeconds: {{ .Values.server.startupProbe.initialDelaySeconds }}
+          successThreshold: {{ .Values.server.startupProbe.successThreshold }}
+          failureThreshold: {{ .Values.server.startupProbe.failureThreshold }}
+          periodSeconds: {{ .Values.server.startupProbe.periodSeconds }}
+          timeoutSeconds: {{ .Values.server.startupProbe.timeoutSeconds }}
+        {{- end }}
+        {{- if .Values.server.readinessProbe.enabled }}
+        readinessProbe:
+          httpGet:
+            path: {{ .Values.server.healthcheckPath }}
+            port: {{ .Values.server.container.port }}
+          initialDelaySeconds: {{ .Values.server.readinessProbe.initialDelaySeconds }}
+          successThreshold: {{ .Values.server.readinessProbe.successThreshold }}
+          failureThreshold: {{ .Values.server.readinessProbe.failureThreshold }}
+          periodSeconds: {{ .Values.server.readinessProbe.periodSeconds }}
+          timeoutSeconds: {{ .Values.server.readinessProbe.timeoutSeconds }}
+        {{- end }}
+        {{- if .Values.server.livenessProbe.enabled }}
+        livenessProbe:
+          httpGet:
+            path: {{ .Values.server.healthcheckPath }}
+            port: {{ .Values.server.container.port }}
+          initialDelaySeconds: {{ .Values.server.livenessProbe.initialDelaySeconds }}
+          successThreshold: {{ .Values.server.livenessProbe.successThreshold }}
+          failureThreshold: {{ .Values.server.livenessProbe.failureThreshold }}
+          periodSeconds: {{ .Values.server.livenessProbe.periodSeconds }}
+          timeoutSeconds: {{ .Values.server.livenessProbe.timeoutSeconds }}
+        {{- end }}
+        resources:
+          {{- toYaml .Values.server.resources | nindent 10 }}
+        volumeMounts:
+        - name: config
+          mountPath: /config
+        {{- if .Values.server.dbDataCm }}
+        - name: imports
+          mountPath: /app/dist/init/db/imports
+        {{- end }}
+        {{- if and .Values.server.plugins (len .Values.server.plugins) }}
+        - name: plugins
+          mountPath: /plugins
+        {{- end }}
+        {{- range $volumeMount := .Values.server.extraVolumeMounts }}
+        - name: {{ $volumeMount.name }}
+          mountPath: {{ $volumeMount.mountPath }}
+        {{- end }}
+      {{- with .Values.server.nodeSelector }}
+      nodeSelector:
```

Toujours utile de laisser la possibilité d'ajouter des sidescars
```suggestion
      {{- if .Values.server.extraContainers }}
        {{- toYaml .Values.server.extraContainers | nindent 8 }}
      {{- end }}
      {{- with .Values.server.nodeSelector }}
      nodeSelector:
```